src/tools/compiletest/src/runtest.rs

```diff
@@ -3726,6 +3737,239 @@ impl<'test> TestCx<'test> {
         fs::remove_dir(path)
     }
 
+    fn run_rmake_v2_test(&self) {
+        // For `run-make` V2, we need to perform 2 steps to build and run a `run-make` V2 recipe
+        // (`rmake.rs`) to run the actual tests. The support library is already built as a tool
+        // dylib and is available under `build/$TARGET/stageN-tools-bin/librun_make_support.so`.
+        //
+        // 1. We need to build the recipe `rmake.rs` and link in the support library.
+        // 2. We need to run the recipe to build and run the tests.
+        let cwd = env::current_dir().unwrap();
+        let src_root = self.config.src_base.parent().unwrap().parent().unwrap();
+        let src_root = cwd.join(&src_root);
+        let build_root = self.config.build_base.parent().unwrap().parent().unwrap();
+        let build_root = cwd.join(&build_root);
+
+        let tmpdir = cwd.join(self.output_base_name());
+        if tmpdir.exists() {
+            self.aggressive_rm_rf(&tmpdir).unwrap();
+        }
+        create_dir_all(&tmpdir).unwrap();
+
+        // HACK: assume stageN-target, we only want stageN.
+        let stage = self.config.stage_id.split('-').next().unwrap();
+
+        // First, we construct the path to the built support library.
+        let mut support_dylib_path = PathBuf::new();
+        support_dylib_path.push(&build_root);
+        support_dylib_path.push(format!("{}-tools-bin", stage));
+        if self.config.target.contains("windows") {
+            support_dylib_path.push("librun_make_support.dll");
+        } else if self.config.target.contains("apple") {
+            support_dylib_path.push("librun_make_support.dylib");
+        } else {
+            support_dylib_path.push("librun_make_support.so");
+        }
+
+        let mut stage_std_path = PathBuf::new();
+        stage_std_path.push(&build_root);
+        stage_std_path.push(&stage);
+        stage_std_path.push("lib");
+
+        // Then, we need to build the recipe `rmake.rs` and link in the support library.
+        let recipe_bin = tmpdir.join("rmake");
+
+        let mut support_dylib_deps = PathBuf::new();
+        support_dylib_deps.push(&build_root);
+        support_dylib_deps.push(format!("{}-tools", stage));
+        support_dylib_deps.push(&self.config.target);
+        support_dylib_deps.push("release/deps");
+
+        let mut support_dylib_deps_deps = PathBuf::new();
+        support_dylib_deps_deps.push(&build_root);
+        support_dylib_deps_deps.push(format!("{}-tools", stage));
+        support_dylib_deps_deps.push("release/deps");
+
+        debug!(?support_dylib_deps);
+        debug!(?support_dylib_deps_deps);
+
+        let res = self.cmd2procres(
+            Command::new(&self.config.rustc_path)
```

Comment:
This misses a `--target` passing the build triple.

Reply:
Not 100% certain on what to pass here, `self.config.host`?

Reply:
On second thought I think leaving it out is fine after all.